### PR TITLE
Run clang-format

### PR DESCRIPTION
**Pull request by marijnvdwerf**

### CODE REVIEW

File: src/7ABB10.c [hunk 1]
```diff
@@ -19,8 +19,10 @@ void func_802D2604(GObj* obj) {
     animal* animal = obj->data.animal;
 
     runGObjProcess(obj, spawnStaryuAtGeo, 1, 1);
-    
+
+    // clang-format off
```
Comment (ethteck) [inline src/7ABB10.c:23]: needing these is un-ideal. can we just not enforce newline rules? Or maybe a leading slash + newline will work here

**Review by ethteck — CHANGES_REQUESTED** [review]
Overall happy with the idea / changes, but I'd prefer to adjust some settings. did you run clang-format with default settings? I can try to mess with things and create a settings file

**Comment by marijnvdwerf** [comment]
I ran this using the clang-format file already in the repo

File: src/46270.c [hunk 2]
```diff
@@ -36,11 +36,13 @@ void func_8009B40C(void) {
         load_overlay(&D_800ABDEC);
         load_overlay(&D_800ABBD0);
         if (func_801DD010(i) != 0) {
-            while (TRUE);
+            while (TRUE)
+                ;
```
[the remark this answers is not in the document]
Comment (marijnvdwerf) [reply src/46270.c:40]: IMO this is more readable than `while (TRUE);`. Below could be another option. Or making an `INFINITE_LOOP()` macro. (The latter probably is the most readable solution)

```
while (TRUE) {
}
```

File: src/47380.c [hunk 3]
```diff
@@ -83,7 +83,7 @@ s32 func_8009CDE4(u8* arg0) {
 
 void func_8009D1E8(u32 arg0, s32 arg1, s32 arg2) {
     if (arg1 >= arg0) {
-        dma_rom_read(arg0, (void*) arg2, arg1 - arg0);
+        dma_rom_read(arg0, (void*)arg2, arg1 - arg0);
```
[the remark this answers is not in the document]
Comment (marijnvdwerf) [reply src/47380.c:86]: [`spaceaftercstylecast`](https://clang.llvm.org/docs/ClangFormatStyleOptions.html#spaceaftercstylecast)

File: include/common_structs.h [hunk 4]
```diff
@@ -393,7 +393,7 @@ typedef struct {
     /* 0x04C */ Vec3f collisionOffset;
     /* 0x058 */ f32 collisionRadius;
     /* 0x05C */ animalAnimationSetup* animSetup; /* Created by retype action */
-    /* 0x060 */ struct GObj* interactionProc; /* Created by retype action */
+    /* 0x060 */ struct GObj* interactionProc;    /* Created by retype action */
```
Comment (marijnvdwerf) [inline include/common_structs.h:396]: I don't like these trailing comments being aligned. Makes for messy diffs imo.

File: include/functions.h [hunk 5]
```diff
@@ -44,7 +44,8 @@ void runAnimalCleanup(GObj* obj);
 GObj* addAnimalAtGeo(GObj* obj, u16 AnimalID, animalDef* def);
 roomGFX* setNodePosToNegRoom(GObj*);
 void animalPathLoop(GObj* obj, f32 start, f32 end, f32 dt, f32 yawStep, u32 flags);
-void spawnAnimalUsingDeltaHeight(s32 gObjID, u16 id, roomGFX* roomA, roomGFX* roomB, objectSpawn* spawn, animalInitData* initData);
+void spawnAnimalUsingDeltaHeight(s32 gObjID, u16 id, roomGFX* roomA, roomGFX* roomB, objectSpawn* spawn,
```
Comment (marijnvdwerf) [inline include/functions.h:47]: Would rather see this like
```
void spawnAnimalUsingDeltaHeight(
    s32 gObjID,
    u16 id,
    roomGFX* roomA,
    roomGFX* roomB,
    objectSpawn* spawn,
    animalInitData* initData
);

File: src/46270.c (hunk 2 above)
[the remark this answers is not in the document]
Comment (ethteck) [reply src/46270.c:40]: I am happy with while TRUE) {
}

File: include/functions.h (hunk 5 above)
Comment (ethteck) [reply include/functions.h:47]: I agree for long arg lists but not in general (not sure if this is configurable based on line length)

File: src/sys/cont.c [hunk 6]
```diff
@@ -36,11 +36,7 @@ enum ControllerDeviceTypes {
     CONT_DEV_TYPE_PRINTER = 5
 };
 
-enum RumbleCommands {
-    RUMBLE_CMD_INIT = 0,
-    RUMBLE_CMD_START = 1,
-    RUMBLE_CMD_STOP = 2
-};
+enum RumbleCommands { RUMBLE_CMD_INIT = 0, RUMBLE_CMD_START = 1, RUMBLE_CMD_STOP = 2 };
```
Comment (marijnvdwerf) [inline src/sys/cont.c:39]: I do _not_ like this

File: src/sys/gtl.c [hunk 7]
```diff
@@ -51,10 +51,12 @@ typedef struct FnBundle {
 s32 gtlD_80040CF0 = 0;
 u32 gtlFrameCounter = 0;
 s32 gtlDrawnFrameCounter = 0;
-UcodeInfo gtlD_80040CFC[] = { { NULL, NULL }, { NULL, NULL }, { gspF3DEX2_NoN_fifoTextStart, gspF3DEX2_NoN_fifoDataStart },
-                              { NULL, NULL }, { NULL, NULL }, { NULL, NULL },
-                              { NULL, NULL }, { NULL, NULL }, { gspL3DEX2_fifoTextStart, gspL3DEX2_fifoDataStart },
-                              { NULL, NULL }, { NULL, NULL } };
+UcodeInfo gtlD_80040CFC[] = {
+    { NULL, NULL }, { NULL, NULL }, { gspF3DEX2_NoN_fifoTextStart, gspF3DEX2_NoN_fifoDataStart },
```
Comment (marijnvdwerf) [inline src/sys/gtl.c:55]: This is how I'd like to see long function arguments being handled as well. All arguments in a block 4 spaces deeper, but with some manual control over the linebreaks, in order to group 'related' properties manually (e.g. x/y).

File: src/sys/gtl.c [hunk 8]
```diff
@@ -772,9 +779,12 @@ void gtlMain(FnBundle* arg0) {
     }
 
     gtlWaitAllGfxTasksDone();
-    while (osRecvMesg(&gtlD_800497E0, NULL, OS_MESG_NOBLOCK) != -1) {}
-    while (osRecvMesg(&gtlResetQueue, NULL, OS_MESG_NOBLOCK) != -1) {}
-    while (osRecvMesg(&gtlGameTickQueue, NULL, OS_MESG_NOBLOCK) != -1) {}
+    while (osRecvMesg(&gtlD_800497E0, NULL, OS_MESG_NOBLOCK) != -1) {
```
Comment (marijnvdwerf) [inline src/sys/gtl.c:782]: Maybe consider a `WAIT_UNTIL( x )` for these empty while loops?

File: src/sys/om.c [hunk 9]
```diff
@@ -1357,7 +1358,8 @@ OMCamera* omGObjSetCamera(GObj* obj) {
     obj->data.cam = camera;
     camera->obj = obj;
     func_80007CBC(&camera->vp);
-    if (FALSE) {} // required to match
+    if (FALSE) {
```
Comment (marijnvdwerf) [inline src/sys/om.c:1361]: These probably would look less weird if we just put the `// required to match` on the line above

File: src/1A560.c [hunk 10]
```diff
@@ -85,8 +85,7 @@ s32 func_80019D54(void) {
 
 u16 func_80019D60(void) {
     // Required to be one line to match.
-    D_80040F40 = (D_80040F40 * 0x343FD) + 0x269EC3; \
-    return (D_80040F40 >> 16);
+    return ((D_80040F40 = (D_80040F40 * 0x343FD) + 0x269EC3) >> 16);
```
Comment (ethteck) [inline src/1A560.c:88]: I'd prefer the version we had prior to this change

File: src/sys/gtl.c (hunk 8 above)
Comment (ethteck) [reply src/sys/gtl.c:782]: I think the new formatting of them is fine

File: src/menu_new_game/A5CC50.c [hunk 11]
```diff
@@ -8,78 +8,358 @@ extern Sprite D_80117F98_A93348; // card
...
-    "ｙ", "ｚ", "０", "１", "２",
-    "３", "４", "５", "６", "７",
-    "８", "９", "†", "‡", "¶",
-    "\xA1\xC1", "！", "？", "＠", "＃",
-    "％", "＆", "＊", "（", "）",
-    "＿", "＋", "＝", "ー", "’",
-    "”", "；", "：", "／", "＄",
-    "．", "，", "＜", "＞", "♂",
-    "♀", "［", "］", "ε", "　",
+    "あ",
```
Comment (ethteck) [inline src/menu_new_game/A5CC50.c:11]: this isn't ideal

Comment (marijnvdwerf) [reply src/menu_new_game/A5CC50.c:11]: This is gonna need `clang-format off` fences, due to the manual formatting.

File: src/world/block.c [hunk 12]
```diff
@@ -622,7 +617,10 @@ WorldBlock** createWorldBlocks(UnkBoneFox* arg0, s32 skyBoxObjId, s32 blockMinOb
         s0->descriptor = ptr[i - num1];
         s0->prev = NULL;
         s0->next = NULL;
-        if (1) { } if (1) { } // TODO fake match
+        if (1) {
```
Comment (marijnvdwerf) [inline src/world/block.c:620]: I think it's good that these get split over multiple lines, but might be worth checking what`AllowShortIfStatementsOnASingleLine: WithoutElse` would do.

**Comment by ethteck** [comment]
gonna close this since we're incrementally running it on every file we work on anyway. thanks for your work on the clang format rules file :)

**Pull request closed without merging**